src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/ContactDiaryStoreV1.swift

```diff
@@ -0,0 +1,749 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+import UIKit
+import FMDB
+import Combine
+
+// swiftlint:disable:next type_body_length
+class ContactDiaryStoreV1: DiaryStoring, DiaryProviding {
+
+	static let encriptionKeyKey = "ContactDiaryStoreEncryptionKey"
+
+	// MARK: - Init
+
+	init(
+		databaseQueue: FMDatabaseQueue,
+		schema: ContactDiaryStoreSchemaV1,
+		key: String
+	) {
+		self.databaseQueue = databaseQueue
+		self.key = key
+
+		openAndSetup()
+
+		createSchemaIfNeeded(schema: schema)
+		_ = cleanup()
+
+		databaseQueue.inDatabase { database in
+			_ = updateDiaryDays(with: database)
+		}
+
+		registerToDidFinishLaunchingNotification()
+	}
+
+	// MARK: - Protocol DiaryProviding
+
+	var diaryDaysPublisher = CurrentValueSubject<[DiaryDay], Never>([])
+
+	// MARK: - Protocol DiaryStoring
+
+	func cleanup() -> DiaryStoringVoidResult {
+		var result: DiaryStoringVoidResult = .success(())
+
+		databaseQueue.inDatabase { database in
+			Log.info("[ContactDiaryStore] Cleanup old entries.", log: .localData)
+
+			guard database.beginExclusiveTransaction() else {
+				Log.error("[ContactDiaryStore] (\(database.lastErrorCode())) \(database.lastErrorMessage())", log: .localData)
+				result = .failure(SQLiteErrorCode(rawValue: database.lastErrorCode()) ?? SQLiteErrorCode.unknown)
+				return
+			}
+
+			let sqlContactPersonEncounter = """
+				DELETE FROM ContactPersonEncounter
+				WHERE date < date('now','-\(dataRetentionPeriodInDays) days')
+			"""
+
+			let sqlLocationVisit = """
+				DELETE FROM LocationVisit
+				WHERE date < date('now','-\(dataRetentionPeriodInDays) days')
+			"""
+
+			do {
+				try database.executeUpdate(sqlContactPersonEncounter, values: nil)
+				try database.executeUpdate(sqlLocationVisit, values: nil)
+			} catch {
+				Log.error("[ContactDiaryStore] (\(database.lastErrorCode())) \(database.lastErrorMessage())", log: .localData)
+				result = .failure(SQLiteErrorCode(rawValue: database.lastErrorCode()) ?? SQLiteErrorCode.unknown)
+				return
+			}
+
+			guard database.commit() else {
+				Log.error("[ContactDiaryStore] (\(database.lastErrorCode())) \(database.lastErrorMessage())", log: .localData)
+				result = .failure(SQLiteErrorCode(rawValue: database.lastErrorCode()) ?? SQLiteErrorCode.unknown)
+				return
+			}
```

Comment:
Don't we need to call `updateDiaryDays(with: database)` here as well to update the views after cleaning up?

Reply:
Very good point! Thanks!